Sources/System/MachPort.swift

```diff
@@ -0,0 +1,327 @@
+/*
+ This source file is part of the Swift System open source project
+
+ Copyright (c) 2022 Apple Inc. and the Swift System project authors
+ Licensed under Apache License v2.0 with Runtime Library Exception
+
+ See https://swift.org/LICENSE.txt for license information
+*/
+
+#if $MoveOnly && (os(macOS) || os(iOS) || os(watchOS) || os(tvOS))
+
+import Darwin.Mach
+
+public protocol MachPortRight {}
+
+private func machPrecondition(
+    file: StaticString = #file,
+    line: UInt = #line,
+    _ body: @autoclosure () -> kern_return_t
+) {
+    let kr = body()
+    let expected = KERN_SUCCESS
+    precondition(kr == expected, file: file, line: line)
+}
+
+public enum Mach {
+    @_moveOnly
+    public struct Port<RightType: MachPortRight> {
+        internal var name: mach_port_name_t
+        internal var context: mach_port_context_t
+
+        /// Transfer ownership of an existing unmanaged Mach port right into a
+        /// Mach.Port by name.
+        ///
+        /// This initializer aborts if name is MACH_PORT_NULL, or if name is
+        /// MACH_PORT_DEAD and the type T of Mach.Port<T> is Mach.ReceiveRight.
+        ///
+        /// If the type of the right does not match the type T of Mach.Port<T>
+        /// being constructed, behavior is undefined.
+        ///
+        /// The underlying port right will be automatically deallocated at the
+        /// end of the Mach.Port instance's lifetime.
+        ///
+        /// This initializer makes a syscall to guard the right.
+        public init(name: mach_port_name_t) {
+            precondition(name != mach_port_name_t(MACH_PORT_NULL), "Mach.Port cannot be initialized with MACH_PORT_NULL")
+            self.name = name
+
+            if RightType.self == ReceiveRight.self {
+                precondition(name != 0xFFFFFFFF /* MACH_PORT_DEAD */, "Receive rights cannot be dead names")
```

Comment:
Out of curiosity, why do we need the constant value instead of the macro as in `MACH_PORT_NULL` above?

Reply:
This is actually an issue with the Darwin Mach overlay, which has this name, but it doesn't work. From what I can tell, the Swift importer couldn't tell this should have been an unsigned value and exposed it as -1, which is not convertible to the unsigned `mach_port_name_t` type.

Reply:
We have the option to add public `Mach.Port.null` and `Mach.Port.dead` names for these, as static computed properties that return `mach_port_name_t` values (or even `Mach.Port`, if that makes sense).